### PR TITLE
Funcionality for tem

### LINKED ISSUE #27: Add BalancingZones to caf.core

### PR DESCRIPTION
**Pull request by isaac-tfn**
Describe Changes
---
Add in any new functionality needed for the new trip end model

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `setup.cfg`

### CODE REVIEW

File: src/caf/core/zoning.py [hunk 1]
```diff
@@ -725,6 +749,231 @@ class ZoningSystemMetaData(ctk.BaseConfig):
...
+            Path to config file, should be the format defined
+            by `configparser` with section names defined in
+            `BalancingZones.OUTPUT_FILE_SECTIONS`.
+
+        Returns
+        -------
+        BalancingZones
+            Balancing zones with loaded parameters.
+        """
+        conf = cls.BalancingConfClass.load(path)
```
Comment (MattBuckley-TfN) [inline src/caf/core/zoning.py:892]: The method is `load_yaml` not `load`
```suggestion
        conf = cls.BalancingConfClass.load_yaml(path)
```

File: src/caf/core/data_structures.py [hunk 2]
```diff
@@ -665,5 +1009,265 @@ def old_to_new_dvec(import_data: dict):
...
+        elif isinstance(balancing_zones, BalancingZones):
+            if balancing_zones._segment_values is not None:
+                if len(balancing_zones._segment_values.keys()) > 1:
+                    # TODO implement this
+                    raise ValueError(
+                        "This method is not currently implemented for "
+                        "balancing zones with individual values defined "
+                        "for multiple segments."
+                    )
+                seg = balancing_zones._segment_values.keys()[0]
```
Comment (MattBuckley-TfN) [inline src/caf/core/data_structures.py:1211]: Should this loop through `_segment_values`?

File: src/caf/core/data_structures.py [hunk 3]
```diff
@@ -665,5 +1009,265 @@ def old_to_new_dvec(import_data: dict):
...
+                    other_remaining,
+                    other.zoning_system,
+                    balancing_zones._default_zoning,
+                )
+                balanced = pd.concat(
+                    [self_slice * slice_factors, self_remaining * remaining_factors]
+                )
+            else:
+                balanced = self.data.copy()
+                for zon, segs in balancing_zones.zoning_groups():
```
Comment (MattBuckley-TfN) [inline src/caf/core/data_structures.py:1233]: Can we chat about this method in more detail?

**Review by MattBuckley-TfN — COMMENTED** [review]
I understand this is still WIP but I've added some initial comments, I think a few might be worth a chat about to understand exactly how the code works.

I can do another review once the PR is ready and look at the tests then.

File: pyproject.toml [hunk 4]
```diff
@@ -20,7 +20,7 @@ classifiers = [
 ]
 dependencies = [
     "numpy>=1.19.0",
-    "pandas<2.0.0" ,
+    "pandas>2.0.0" ,
```
Comment (BenTaylor-TfN) [inline pyproject.toml:23]:
```suggestion
    "pandas>=2.0.0" ,
```

Fixes #1, make sure requirements.txt is updated too

File: requirements.txt [hunk 5]
```diff
@@ -1,8 +1,8 @@
 numpy>=1.24.3
-pandas<2.0
+pandas>=2.0
```
Comment (BenTaylor-TfN) [inline requirements.txt:2]: Fixes #1, make sure pyproject.toml is updated too
